transcom/mymove
# B-21170-INT - Location Lookup for Prime Sim (Review After B-21168-INT & B-21169-INT)

## [B-21170](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21170)

## Summary

Add's the Zip/City/State Finder to all address forms in the prime simulator.

Includes all changes from B-21168 and B-21169, best to wait to review until that integration PR is merged.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

We have ~5~ 6 forms to verify use the city finder and use it correctly.

1. Access the office app
2. Login as a prime simulator
3. Select any move
4. For each of the following select a city for all city finders and ensure all fields populate and save correctly:
5. Form 1 - Add a PPM shipment with secondary and tertiary addresses of both kinds (the backend seemed to be a bit finicky with the secondary and tertiary PPM addresses but that's out of scope)
6. Form 2 - Modify that same PPM shipment being sure to change the addresses available to you
7. Form 1 (but for non-PPMs) - Create an HHG shipment with secondary and tertiary addresses of both kinds
8. Form 3 - Modify the pick up address of that HHG
9. Form 3 - With the same form, modify the destination address for that unapprove HHG
11. Form 4 - Identify an existing approved HHG, modify the destination address using the "Update Shipment Destination Address" form.
12. Optionally, access that same HHG as a TOO and approve the prime's request to change the address then verify the approved change displays correctly in the prime sim.
13. Form 5 - Identify an approved HHG that does not have an Origin SIT service item, add one.
14. **NEW** Form 6 - Identify an approved HHG that does not have secondary or tertiary pickup or destination addresses (most e2e populate moves fit the bill). Update the shipment using "Update Shipment." You should be able to add secondary and tertiary pickup and destination addresses **NEW**

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
